pep-0722.rst

```diff
@@ -0,0 +1,563 @@
+dependencies in the script's docstring.
+
+The most significant problem with this proposal is that it requires all
+consumers of the dependency data to implement a Python parser. Even if the
+syntax is restricted, the *rest* of the script will use the full Python syntax,
+and trying to define a syntax which can be successfully parsed in isolation from
+the surrounding code is likely to be extremely difficult and error-prone.
+
+Furthermore, Python's syntax changes in every release. If extracting dependency
+data needs a Python parser, the parser will need to know which version of Python
+the script is written for, and the overhead for a generic tool of having a
+parser that can handle *multiple* versions of Python is unsustainable.
+
+Even if the above issues could be addressed, the format would give the
+impression that the data could be altered at runtime. However, this is not the
+case in general, and code that tries to do so will encounter unexpected and
+confusing behaviour.
+
+And finally, there is no evidence that having dependency data available at
+runtime is of any practical use. Should such a use be found, it is simple enough
+to get the data by parsing the source - ``read_dependency_block(__file__)``.
+
+It is worth noting, though, that the ``pip-run`` utility does implement (an
+extended form of) this approach. `Further discussion <pip-run issue_>`_ of
+the ``pip-run`` design is available on the project's issue tracker.
+
+Should scripts be able to specify a package index?
+--------------------------------------------------
+
+Dependency metadata is about *what* package the code depends on, and not *where*
+that package comes from. There is no difference here between metadata for
+scripts, and metadata for distribution packages (as defined in
+``pyproject.toml``). In both cases, dependencies are given in "abstract" form,
+without specifying how they are obtained.
+
+Some tools that use the dependency information may, of course, need to locate
+concrete dependency artifacts - for example if they expect to create an
+environment containing those dependencies. But the way they choose to do that
+will be closely linked to the tool's UI in general, and this PEP does not try to
+dictate the UI for tools.
+
+There is more discussion of this point, and in particular of the UI choices made
+by the ``pip-run`` tool, in `the previously mentioned pip-run issue <pip-run
+issue_>`_.
+
+What about local dependencies?
+------------------------------
+
+These can be handled without needing special metadata and tooling, simply by
+adding the location of the dependencies to ``sys.path``. This PEP simply isn't
+needed for this case. If, on the other hand, the "local dependencies" are actual
+distributions which are published locally, they can be specified as usual with a
+:pep:`508` requirement, and the local package index specified when running a
+tool by using the tool's UI for that.
+
+Open Issues
+===========
+
+None at this point.
+
+
+Footnotes
```

Comment:
We automatically remove 'references' sections with no content bar targets.

```suggestion
References
```

Reply:
I don’t understand this comment sorry. Footnotes was the header in the template. But I’m fine with changing it if you think it helps.

Reply:
TBH, it's not that important. I certainly don't have any problem with accepting this suggestion, and I'd rather get this merged sooner rather than later, so that the PEP is available "for real", rather than just as a PR. So I'll commit this suggestion and then merge once CI passes. Hope that's OK.

Reply:
My understanding: if this section is named References and contains only reST link targets, then the data will be used to create the link elements and the section will be removed from the rendered contents, instead of being there and looking strange (since PEPs now use real links and not paper-style `[1]` references that lead to end of document)

Reply:
Yeah, just spotted that, thanks 🙂 And thanks for the explanation.

Reply:
Sorry for being a bit too brief -- @merwork is exactly right!

A